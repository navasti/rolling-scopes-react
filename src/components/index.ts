export { Loader } from './Loader';
export { Modal } from './Modal';
