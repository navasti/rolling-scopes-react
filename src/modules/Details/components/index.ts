export { MoveCardDetails } from './MoveCardDetails';
export { TypeCardDetails } from './TypeCardDetails';
export { PokemonCardDetails } from './PokemonCardDetails';
