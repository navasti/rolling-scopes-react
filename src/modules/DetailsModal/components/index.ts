export { TypeModalContent } from './TypeModalContent';
export { MoveModalContent } from './MoveModalContent';
export { PokemonModalContent } from './PokemonModalContent';
