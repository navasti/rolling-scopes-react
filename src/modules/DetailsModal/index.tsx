import { PokemonDetails, PokemonMoveDetails, PokemonTypeDetails } from 'types';
import { Modal } from 'components';
import { forwardRef } from 'react';
import * as S from './styled';
import {
  PokemonModalContent,
  MoveModalContent,
  TypeModalContent,
} from 'modules/DetailsModal/components';

type Props = {
  selectedType?: PokemonTypeDetails;
  selectedMove?: PokemonMoveDetails;
  selectedPokemon?: PokemonDetails;
  handleCloseModal: () => void;
  isModalOpened: boolean;
};

export const DetailsModal = forwardRef<HTMLDivElement, Props>((props, ref) => {
  const { handleCloseModal, isModalOpened, selectedMove, selectedPokemon, selectedType } = props;
  return (
    <>
      {selectedPokemon ? (
        <Modal handleCloseModal={handleCloseModal} opened={isModalOpened} ref={ref}>
          <S.ModalTitle>{selectedPokemon.name}</S.ModalTitle>
          <PokemonModalContent selectedPokemon={selectedPokemon} />
        </Modal>
      ) : selectedMove ? (
        <Modal handleCloseModal={handleCloseModal} opened={isModalOpened} ref={ref}>
          <S.ModalTitle>{selectedMove.name}</S.ModalTitle>
          <MoveModalContent selectedMove={selectedMove} />
        </Modal>
      ) : selectedType ? (
        <Modal handleCloseModal={handleCloseModal} opened={isModalOpened} ref={ref}>
          <S.ModalTitle>{selectedType.name}</S.ModalTitle>
          <TypeModalContent selectedType={selectedType} />
        </Modal>
      ) : null}
    </>
  );
});
