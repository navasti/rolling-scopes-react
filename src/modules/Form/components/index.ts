export { BirthdayField } from './BirthdateField';
export { ErrorMessage } from './ErrorMessage';
export { ConsentField } from './ConsentField';
export { AvatarField } from './AvatarField';
export { GenderField } from './GenderField';
export { ShinyField } from './ShinyField';
export { TypeField } from './TypeField';
export { NameField } from './NameField';
export { FormCard } from './FormCard';
