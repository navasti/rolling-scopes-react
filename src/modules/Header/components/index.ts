export { LocationBadge } from './LocationBadge';
export { Navigation } from './Navigation';
