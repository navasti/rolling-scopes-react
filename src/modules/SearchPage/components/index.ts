export { SearchBar } from './SearchBar';
export { Cards } from './Cards';
export { Card } from './Card';
export { Tabs } from './Tabs';
