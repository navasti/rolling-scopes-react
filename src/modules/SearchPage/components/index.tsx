export { SearchBar } from './SearchBar';
export { Card } from './Card';
