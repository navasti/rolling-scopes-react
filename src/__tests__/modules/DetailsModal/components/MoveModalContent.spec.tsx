import { MoveModalContent } from 'modules/DetailsModal/components';
import { screen, render } from '@testing-library/react';

describe('', () => {
  it('', () => {
    render(<MoveModalContent selectedMove={} />);
  });
});
