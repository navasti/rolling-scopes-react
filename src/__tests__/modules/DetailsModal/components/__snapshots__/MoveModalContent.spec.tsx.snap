// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoveModalContent content should match the snapshot with passed first element of movesMock as a prop 1`] = `
<div>
  <div>
    <p
      class="sc-gsnTZi hZMIDi"
    >
      <b>
        Accuracy
        :
      </b>
       
      100
    </p>
    <p
      class="sc-gsnTZi hZMIDi"
    >
      <b>
        Type
        :
      </b>
       
      normal
    </p>
    <p
      class="sc-gsnTZi hZMIDi"
    >
      <b>
        Power
        :
      </b>
       
      40
    </p>
    <p
      class="sc-gsnTZi hZMIDi"
    >
      <b>
        Priority
        :
      </b>
       
      0
    </p>
    <p
      class="sc-gsnTZi hZMIDi"
    >
      <b>
        PP
        :
      </b>
       
      35
    </p>
    <p
      class="sc-gsnTZi hZMIDi"
    >
      <b>
        Learned by:
      </b>
       
      bulbasaur
    </p>
  </div>
</div>
`;
